shards/shards.go

```diff
@@ -749,6 +749,8 @@ func copyFiles(sr *zoekt.SearchResult) {
 		copySlice(&sr.Files[i].Checksum)
 		for l := range sr.Files[i].LineMatches {
 			copySlice(&sr.Files[i].LineMatches[l].Line)
+			copySlice(&sr.Files[i].LineMatches[l].Before)
+			copySlice(&sr.Files[i].LineMatches[l].After)
```

Comment:
minor: can you specialize copySlice to be a noop when the input is nil. Right now you will allocate an empty slice with this call, even though NumContextLines is not set.